instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/HttpServerRequestMetrics.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.AttributeType;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleValueRecorder;
+import io.opentelemetry.api.metrics.LongUpDownCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.api.metrics.common.LabelsBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class HttpServerRequestMetrics implements RequestMetrics {
+
+  private static final ContextKey<State> HTTP_SERVER_REQUEST_METRICS_STATE =
+      ContextKey.named("http-server-request-metrics-state");
+
+  private static final Logger logger = LoggerFactory.getLogger(HttpServerRequestMetrics.class);
+
+  public static RequestMetricsFactory factory() {
+    return HttpServerRequestMetrics::new;
+  }
+
+  private final LongUpDownCounter activeRequests;
+  private final DoubleValueRecorder duration;
+
+  private HttpServerRequestMetrics(Meter meter) {
+    activeRequests =
+        meter
+            .longUpDownCounterBuilder("http.server.active_requests")
+            .setUnit("requests")
```

Comment:
This makes me think the semantic conventions need to have units in them as well, so we won't have to hard-code this stuff. Also, even though this is currently the semantic convention for these units, I suspect this will not last, as we will probably be standardizing on units in the future. See this issue for tracking: https://github.com/open-telemetry/opentelemetry-specification/issues/705

Reply:
I was thinking the generator should provide them as well. Are they in the yaml yet?

Reply:
Ah got it - nope there aren't any metric yaml files yet